src/plugins/section_title/autocompletion.ts

```diff
@@ -69,23 +63,25 @@ export function checkForCompletion(state: EditorState) {
     state.selection.$from,
     schema.nodes.section
   )
+  if (!section) {
+    return
+  }
+
   const title = findParentNodeOfTypeClosestToPos(
     state.selection.$from,
     schema.nodes.section_title
   )
+  if (!title || !cursorAtTheEndOfText(state, title.node.nodeSize, title.pos)) {
+    return
+  }
 
-  if (
-    section &&
-    title &&
-    cursorAtTheEndOfText(state, title.node.nodeSize, title.pos)
-  ) {
-    const text = hasAutoCompletionSlack(
-      section.node as SectionNode,
-      title.node as SectionTitleNode
-    )
+  const props = getEditorProps(state)
+  const category = props.sectionCategories.get(section.node.attrs.category)
```

Comment:
why do we need to pass categories from props? can they not just be imported from transform?

Reply:
No, the categories will no longer be defined in the transform.

Reply:
Why not have them in transform since api depends on the transform anyway and we can be more aware of the section on frontend too that way?

Reply:
We want to be able to configure them per template

Reply:
ok, but we will still have a finite list of categories? And here it looks like we check whether this category is in the list of available categories for the current template and then we proceed to autocomplete the title. So technically we check if the category of the current section is valid for the current template? Is it possible for a manuscript to include a section that is not allowed for the template?

Reply:
You probably meant by "configure" to be able to change properties of categories including titles etc. That makes sense then. Thanks